utils/config.ts

```diff
@@ -0,0 +1,166 @@
+    t.intersection([
+      t.interface({
+        MAILHOG_HOSTNAME: t.undefined,
+        MAIL_TRANSPORTS: t.undefined,
+        NODE_ENV: t.literal("production"),
+        SENDGRID_API_KEY: NonEmptyString
+      }),
+      t.partial({
+        MAILUP_SECRET: NonEmptyString,
+        MAILUP_USERNAME: NonEmptyString
+      })
+    ]),
+    // using mailup
+    t.interface({
+      MAILHOG_HOSTNAME: t.undefined,
+      MAILUP_SECRET: NonEmptyString,
+      MAILUP_USERNAME: NonEmptyString,
+      MAIL_TRANSPORTS: t.undefined,
+      NODE_ENV: t.literal("production"),
+      SENDGRID_API_KEY: t.undefined
+    }),
+    // Using multi-transport definition
+    // Optional multi provider connection string
+    // The connection string must be in the format:
+    //   [mailup:username:password;][sendgrid:apikey:;]
+    // Note that multiple instances of the same provider can be provided.
+    t.interface({
+      MAILHOG_HOSTNAME: t.undefined,
+      MAILUP_SECRET: t.undefined,
+      MAILUP_USERNAME: t.undefined,
+      MAIL_TRANSPORTS: MailMultiTransportConnectionsFromString,
+      NODE_ENV: t.literal("production"),
+      SENDGRID_API_KEY: t.undefined
+    }),
+    // the following states that a mailhog configuration is optional and can be provided only if not in prod
+    t.interface({
+      MAILHOG_HOSTNAME: NonEmptyString,
+      MAILUP_SECRET: t.undefined,
+      MAILUP_USERNAME: t.undefined,
+      MAIL_TRANSPORTS: t.undefined,
+      NODE_ENV: AnyBut("production", t.string),
+      SENDGRID_API_KEY: t.undefined
+    })
+  ])
+]);
+
+// global app configuration
+export type IConfig = t.TypeOf<typeof IConfig>;
+export const IConfig = t.intersection([
+  t.interface({
+    COSMOSDB_KEY: NonEmptyString,
+    COSMOSDB_NAME: NonEmptyString,
+    COSMOSDB_URI: NonEmptyString,
+
+    SERVICE_PRINCIPAL_CLIENT_ID: NonEmptyString,
+    SERVICE_PRINCIPAL_SECRET: NonEmptyString,
+    SERVICE_PRINCIPAL_TENANT_ID: NonEmptyString,
+
+    AZURE_APIM: NonEmptyString,
+    AZURE_APIM_HOST: NonEmptyString,
+    AZURE_APIM_RESOURCE_GROUP: NonEmptyString,
+    AZURE_SUBSCRIPTION_ID: NonEmptyString,
+
+    ADB2C_CLIENT_ID: NonEmptyString,
+    ADB2C_CLIENT_KEY: NonEmptyString,
+    ADB2C_TENANT_ID: NonEmptyString,
+
+    UserDataBackupStorageConnection: NonEmptyString,
+
+    MESSAGE_CONTAINER_NAME: NonEmptyString,
+    USER_DATA_BACKUP_CONTAINER_NAME: NonEmptyString,
+    USER_DATA_CONTAINER_NAME: NonEmptyString,
+
+    StorageConnection: NonEmptyString,
+    SubscriptionFeedStorageConnection: NonEmptyString,
+    UserDataArchiveStorageConnection: NonEmptyString,
+
+    PUBLIC_API_KEY: NonEmptyString,
+    PUBLIC_API_URL: NonEmptyString,
+
+    PUBLIC_DOWNLOAD_BASE_URL: NonEmptyString,
+
+    SESSION_API_KEY: NonEmptyString,
+    SESSION_API_URL: NonEmptyString,
+
+    LOGOS_URL: NonEmptyString,
+
+    SUBSCRIPTIONS_FEED_TABLE: NonEmptyString,
+    USER_DATA_DELETE_DELAY_DAYS: NonEmptyString,
```

Comment:
We have to remember that in case of a global env variable creation or deletion or renaming we have to update this

Reply:
No problem in creation - you'll be forced to add it to the model. When deleting it is tricky, and we'd must rely on our discipline imho